tests/unit/igraph_random_walk.c

```diff
@@ -19,67 +19,216 @@
 #include <igraph.h>
 #include "test_utilities.h"
 
-void random_walk(igraph_t *graph, igraph_integer_t start, igraph_neimode_t mode,
+void random_vertex_walk(igraph_t *graph, igraph_vector_t *weights,
+    igraph_integer_t start, igraph_neimode_t mode,
     igraph_integer_t steps, igraph_random_walk_stuck_t stuck) {
-    igraph_vector_int_t walk;
+    igraph_vector_int_t vertices;
 
-    igraph_vector_int_init(&walk, 0);
-    igraph_random_walk(graph, &walk, start, mode, steps, stuck);
-    print_vector_int(&walk);
-    igraph_vector_int_destroy(&walk);
+    igraph_vector_int_init(&vertices, 0);
+    igraph_random_walk(graph, weights, &vertices, NULL, start, mode, steps, stuck);
+    print_vector_int(&vertices);
+    igraph_vector_int_destroy(&vertices);
 }
 
-int main() {
-    igraph_t g_1, g_line, g_full, g_loop;
-    igraph_vector_int_t walk;
-
-    igraph_vector_int_init(&walk, 0);
+void random_walk(igraph_t *graph, igraph_vector_t *weights,
+    igraph_integer_t start, igraph_neimode_t mode,
+    igraph_integer_t steps, igraph_random_walk_stuck_t stuck) {
+    igraph_vector_int_t vertices;
+    igraph_vector_int_t edges;
+
+    igraph_vector_int_init(&vertices, 0);
+    igraph_vector_int_init(&edges, 0);
+    igraph_random_walk(graph, weights, &vertices, &edges, start, mode, steps, stuck);
+    print_vector_int(&vertices);
+    igraph_vector_int_destroy(&vertices);
+    igraph_vector_int_destroy(&edges);
+}
 
+int main() {
+    igraph_t g_1, g_line, g_full, g_loop, graph;
+    igraph_vector_int_t vertices, edges;
+    igraph_vector_t g_line_weights, graph_weights, error_weights;
+    igraph_integer_t ec, i;
+
+    igraph_vector_int_init(&vertices, 0);
+    igraph_vector_int_init(&edges, 0);
+    igraph_vector_init(&g_line_weights, 0);
+    igraph_vector_init(&graph_weights, 0);
+    igraph_vector_init(&error_weights, 0);
+    igraph_rng_seed(igraph_rng_default(), 42);
+    
     igraph_small(&g_1, 1, IGRAPH_UNDIRECTED, -1);
     igraph_small(&g_line, 5, IGRAPH_DIRECTED, 0,1, 1,2, 2,3, 3,4, -1);
     igraph_full(&g_full, 20, IGRAPH_UNDIRECTED, 0);
     igraph_small(&g_loop, 1, IGRAPH_DIRECTED, 0,0, -1);
 
+    /* This directed graph has loop edges.
+       It also has multi-edges when considered as undirected. */
+    igraph_de_bruijn(&graph, 3, 2);
+
+    /* Initialize de bruijn graph weights */
+    ec = igraph_ecount(&graph);
+    igraph_vector_resize(&graph_weights, ec);
+    for (i = 0; i < ec; ++i) {
+        VECTOR(graph_weights)[i] = igraph_rng_get_unif01(igraph_rng_default());
+    }
+    
+    /* Initialize g_line graph weights */
+    ec = igraph_ecount(&g_line);
+    igraph_vector_resize(&g_line_weights, ec);
+    for (i = 0; i < ec; ++i) {
+        VECTOR(g_line_weights)[i] = igraph_rng_get_unif01(igraph_rng_default());
+    }
+
+
+    /* Length of the vertices vector -> (steps + 1) -> includes starting and ending vertices of the walk,
```

Comment:
I would remove these comments, they are not really clarifying anything.

Reply:
Sure, removed these comments.